core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/geodistance/GeoDistanceAggregationBuilder.java

```diff
@@ -82,6 +82,23 @@ public GeoDistanceAggregationBuilder(StreamInput in) throws IOException {
         unit = DistanceUnit.readFromStream(in);
     }
 
+    // for parsing
+    GeoDistanceAggregationBuilder(String name) {
```

Comment:
I wanted to do it, but the issue is that the name is not found out while parsing the aggregation but propagated from the aggregation parser of the parent aggregation. We might be able to do that once all aggs use ObjectParser and named objects, but for now I don't think it is possible?